### DISCUSSION TITLE
Compiling NPM modules to Wasm for use in Rust web frameworks (Leptos)

### DISCUSSION THREAD

**Discussion by aschrijver** [WasmEdge/WasmEdge#2607] [Q&A] [answered]
When looking into the docs, WasmEdge is having [broad Javascript support](https://wasmedge.org/docs/category/develop-wasm-apps-in-javascript). I would like to have a full-stack Rust-based project (wasm on server and browser) and for browser-side use [Leptos](https://leptos.dev) to develop UI components in Rust. However, there's the great [React Flow](https://reactflow.dev) component I'd like to integrate with.

The Leptos maintainer [pointed to](https://github.com/leptos-rs/leptos/discussions/1206#discussioncomment-6209401) `wasm-bindgen` as a way to integrate React Flow, but this would involve hand-coding the elaborate API binding in Rust.

What is unclear to me is if WasmEdge is suitable as an alternative to [`wasm-bindgen`](https://github.com/rustwasm/wasm-bindgen) to compile a React Flow `.wasm` usable for Leptos to call into.

**Comment by juntao** [comment, marked as answer]
Since WasmEdge is not bundled in a browser, you cannot use our JS runtime in the browser. The WasmEdge JS runtime is very much geared toward the server side use case -- eg support for node.js APIs and NPM packages on the server. 

In the browser, you will need to use Wasm runtimes like v8 (for chrome) and use wasm-bindgen to facilitate interaction between the Wasm runtime and the browser's JS host.

**Reply by aschrijver** [reply]
Thank you!
